## 第一章：程序设计基本方法

### Q&A
解决Turtle库下载失败问题：https://zhuanlan.zhihu.com/p/114712558

如何在Anaconda下载库：https://blog.csdn.net/kaever/article/details/106295926

### Before the Tutorial
零基础，应当如何开始学习 Python ？

1. 找到合适的入门书籍与教程：首先把本课程教材读懂，不懂就百度，太难的内容可以跳过，然后找到推荐书目或网上教程进行辅助理解。
2. 做好每次实验：Practice makes perfect.练习对于学好一门语言十分重要，不会就反复练习。
3. 保持兴趣，学习解决问题：兴趣是最好的老师，尽量用简单的方式解决问题，减少时间复杂度。
4. 学会检索： 大部分问题都能在博客中找到解决方案，遇到问题先百度。

#### 推荐书目：
* 《笨办法学 Python （Learn Python The Hard Way）》
* 《像计算机科学家一样思考Python》

#### 推荐课程:
Python语言程序设计 北京理工大学 ：https://www.icourse163.org/course/BIT-268001#/info

Python语言程序设计 北京理工大学_B站录制版本：https://www.bilibili.com/video/00000BV1gJ411Q7My?from=search&seid=16962515799502322601

Python123平台： https://python123.io/

### Tutorial - 接下来进入Python的奇妙世界

#### 第一行代码

Python是脚本语言 + 语句执行，我们本此实验使用Python 3.0+ 版本，其版本已经把 print 作为一个内置函数。

如果你是在linux上脚本运行，一定要加上#!/usr/bin/python3

In [1]:
#!/usr/bin/python3
print ("Hello World")

Hello World


Python中默认的编码格式是 ASCII 格式，在没修改编码格式时无法正确打印汉字，所以在读取中文时会报错。

解决方法为只要在文件开头加入 # -*- coding: UTF-8 -*- 或者 # coding=utf-8 就行了

但我们用的是Jupyter Notebook,所以可以省略#!/usr/bin/python3 和# -*- coding: UTF-8 -*-

In [2]:
#!/usr/bin/python3
# -*- coding: UTF-8 -*-
print ('Hello World-三体世界真的存在吗？') 
#Python 不支持单字符类型，单字符在 Python 中也是作为一个字符串使用。

Hello World-三体世界真的存在吗？


In [3]:
print("糊弄学精品词：确实、啊这、是的...")

糊弄学精品词：确实、啊这、是的...


我们也可以在print的输出字符内容里面加上一些转义符

\n 是换行符，通常在输出中用作格式控制；

\n就是一个转义字符，其意义是“回车换行”;

转义字符以反斜线\开头，后跟一个或几个字符。

In [4]:
print("糊弄学精品词：\n确实\n啊这\n是的\n...大家学会了吗")

糊弄学精品词：
确实
啊这
是的
...大家学会了吗


#### Python的简洁 + 强制可读性，你甚至可以把其当作一个计算器

In [5]:
2 + 2

4

In [6]:
(50 - 5.0*6) / 4

5.0

2的7次方是多少？

In [7]:
2 ** 7 

128

你可以把数值赋值给变量（variable），如计算宽和高的乘积

In [8]:
width = 20 # 赋值20给witdth
height = 5 * 9 # 赋值 5 * 9 的值给height
width * height

900

你可以把变量打印出来验证一下

In [9]:
sum = 99999 * 99999
print(sum)

9999800001


In [10]:
width = 20 # 赋值20给witdth
height = 5 * 9 # 赋值 5 * 9 的值给height
print("width is ", width)
print("height is ", height)
print(width * height)

width is  20
height is  45
900


不仅仅可以计算整数，我们还可以计算浮点数，具有混合类型操作数的运算符，会将整数操作数转换为浮点数

In [11]:
3 * 3.75 / 1.5

7.5

In [12]:
months="JanFebMarAprMayJunJulAugSepOctNovDec" 
n = 4
monthAbbrev = months[(n-1)*3:(n-1)*3+3]
print(monthAbbrev)

Apr


In [13]:
def mean(numbers):
    s = 0.0
    for num in numbers:
        s = s + num
    return s / len(numbers)
nums = [0,1,2,3,4,5,6,7,8,9]
print(mean(nums))


4.5


#### Python有着丰富的库
你可以使用保留字import，和保留字组合from...import...分别来导入你想要函数库。
import也可以其他python文件（称为模块module），使用该模块里定义的类、方法或者变量，从而达到代码复用的目的。这是之后进阶的内容。

比如

math --- 数学函数

random --- 生成伪随机数

In [14]:
#import module_name
import random
#from random import random
rnd = random.random()*10
print(rnd)

5.617173900954542


In [15]:
from random import random
rnd = random()*10
print(rnd)

6.78291095457358


In [16]:
import math
math.pi

3.141592653589793

记保留字真的好难，如果你想查看保留字，可以这样做，Python 包含的保留字可以执行如下命令进行查看：

In [17]:
import keyword
keyword.kwlist
print(keyword.kwlist)

['False', 'None', 'True', 'and', 'as', 'assert', 'async', 'await', 'break', 'class', 'continue', 'def', 'del', 'elif', 'else', 'except', 'finally', 'for', 'from', 'global', 'if', 'import', 'in', 'is', 'lambda', 'nonlocal', 'not', 'or', 'pass', 'raise', 'return', 'try', 'while', 'with', 'yield']


#### 海龟绘图教学将在第二章系统学到

在1966年，Seymour Papert和Wally Feurzig发明了一种专门给儿童学习编程的语言——LOGO语言，它的特色就是通过编程指挥一个小海龟（turtle）在屏幕上绘图。

海龟绘图（Turtle Graphics）后来被移植到各种高级语言中，Python内置了turtle库，基本上100%复制了原始的Turtle Graphics的所有功能。

我们来看一个指挥小海龟绘制一个长方形的简单代码：

In [ ]:
# 导入turtle包的所有内容:
from turtle import *

# 设置笔刷宽度:
width(4)

# 前进:
forward(200)
# 右转90度:
right(90)

# 笔刷颜色:
pencolor('red')
forward(100)
right(90)

pencolor('green')
forward(200)
right(90)

pencolor('blue')
forward(100)
right(90)

# 调用done()使得窗口等待被关闭，否则将立刻关闭窗口:
done()

从程序代码可以看出，海龟绘图就是指挥海龟前进、转向，海龟移动的轨迹就是绘制的线条。要绘制一个长方形，只需要让海龟前进、右转90度，反复4次。

调用width()函数可以设置笔刷宽度，调用pencolor()函数可以设置颜色。更多操作请参考turtle库的说明。

绘图完成后，记得调用done()函数，让窗口进入消息循环，等待被关闭。否则，由于Python进程会立刻结束，将导致窗口被立刻关闭。

turtle包本身只是一个绘图库，但是配合Python代码，就可以绘制各种复杂的图形。

具体方法请查阅官方文档：https://docs.python.org/zh-cn/3/library/turtle.html

## 微实例

#### 【微实例1.1】圆面积的计算

In [18]:
radius = 25
area = 3.1415 * radius * radius
print(area)
print("{:.2f}".format(area))

1963.4375000000002
1963.44


#### 扩展练习：关于圆的计算

使用math库也可以计算圆周率，且更加方便了，观察以下例子。

input()返回的的数据类型是str,要进行数学运算，需要用int()或float()函数，转换成整型或浮点型的数据类型。

计算出圆的周长、以该圆半径为半径的球的表面积和体积。补充代码

In [20]:
import math
r=float(input("请输入半径："))
#circumference= #圆的周长
area=math.pi*r*r
#sphere_area= #球的表面积
#sphere_volume= #球的体积
#print ( "圆的周长: %.2f" % circumference)
print ( "圆的面积: %.2f"% area)
#print ( "球的表面积: %.2f"% sphere_area)
#print ( "球的体积: %.2f" % sphere_volume)

请输入半径：1
圆的面积: 3.14


#### 【微实例1.2】简单的人名对话

In [21]:
name = input("输入姓名:")
print("%s 同学，学好Python，前途无量！"%name)
print("%s 大侠，学好Python，大展拳脚！"%name[0])
print("%s 哥哥，学好Python，人见人爱！"%name[1:])

输入姓名:令狐冲
令狐冲 同学，学好Python，前途无量！
令 大侠，学好Python，大展拳脚！
狐冲 哥哥，学好Python，人见人爱！


#### 思考：对于两字或者四字姓名的人名我们该怎么办？英文姓名怎么办？

#### 【微实例1.3】斐波那契数列的计算

斐波那契数列（Fibonacci Sequence )，又称为黄金分割数列，由意大利数学 家 Leonardo Fibonacci 于 1202 年提出，并以其名宇命名。该数列 $\mathrm{F}(n)$ 定义如下:
$\mathrm{F}(0)=0, \mathrm{~F}(1)=1, \mathrm{~F}(n)=\mathrm{F}(n-2)+\mathrm{F}(n-1),$ 其中 $n \geqslant 2 。$ 简单说， 斐波那契数列中每个
数是前两个数之和。斐波那契数列中邻近两个数的比值接近黄金分割数，即 $\mathrm{F}(n) / \mathrm{F}(n-1)$ 接近 $1.618,$ 这个比例的极限值就是黄金分割数。此外，斐波那契数列 有很多特性，所以其在搜索算法、组合数学、现代物理、化学等领域均有应用。 Python 语言在表达和计算传统数学概念上十分简洁，读者可以寻找其他有趣的数
学概念，并用 Python 语言计算它们。

In [22]:
a, b = 0, 1
while a < 1000: # 输出不大于1000 的序列
    print(a, end=',')
    a, b = b, a + b

0,1,1,2,3,5,8,13,21,34,55,89,144,233,377,610,987,

#### 【微实例1.4】同切圆的绘制

Turtle库是Python语言中一个很流行的绘制图像的函数库，想象一个小乌龟，在一个横轴为x、纵轴为y的坐标系原点，(0,0)位置开始，它根据一组函数指令的控制，在这个平面坐标系中移动，从而在它爬行的路径上绘制了图形。

turtle.pensize()：设置画笔的宽度；

turtle.circle()：画圆，半径为正(负)，表示圆心在画笔的左边(右边)画圆

In [23]:
import turtle # 引用turtle 库
turtle.pensize(2) # 设置画笔宽度为2 像素
turtle.circle(10) # 绘制半径为10 像素的圆
turtle.circle(40) # 绘制半径为40 像素的圆
turtle.circle(80) # 绘制半径为80 像素的圆
turtle.circle(160) # 绘制半径为160 像素的圆


#### 扩展练习：尝试不同宽度和半径，如果宽度和半径是浮点数会怎么样？如果两个数字分别是负数会怎么样？如果你在上述代码后面再加几行不同半径不同宽度的代码，会出现什么情况？

#### 【微实例1.5】日期和时间的输出

In [24]:
from datetime import datetime # 引用datetime 库
now = datetime.now() # 获得当前日期和时间信息
print(now)
now.strftime("%x") # 输出其中的日期部分
now.strftime("%X") # 输出其中的时间部分

2021-03-20 19:30:12.225349


'19:30:12'

扩展训练：查阅https://docs.python.org/zh-cn/3/library/datetime.html

或者百度

以now.strftime()函数看出现在的时间是AM还是PM

In [25]:
from datetime import datetime
#请补充

#### 【微实例1.6】字符串拼接 
字符串拼接。接收用户输入的两个字符串，将它们组合后输出。

In [26]:
str1 = input("请输入一个人的名字：")
str2 = input("请输入一个国家名字：")
print("世界这么大，{}想去{}看看。".format(str1,str2))

请输入一个人的名字：艾伦
请输入一个国家名字：马莱
世界这么大，艾伦想去马莱看看。


还记得format函数的作用吗？str.format()是一种格式化字符串的函数

In [27]:
print("{} {}".format("hello", "world"))    # 不设置指定位置，按默认顺序

print("{0} {1}".format("hello", "world"))  # 设置指定位置

print("{1} {0} {1}".format("hello", "world"))  # 设置指定位置

hello world
hello world
world hello world


#### 【微实例1.7】整数序列求和
整数序列求和。用户输入一个正整数 $N$, 计算从 1 到 $N$ (包含 1 和 $N$ ) 相 加之后的结果。

In [28]:
n = input("请输入整数N: ")
sum = 0
for i in range(int(n)):
    sum += i + 1
print("1 到N 求和结果: ", sum)

请输入整数N: 123
1 到N 求和结果:  7626


#### 扩展练习：N=12345的时候的结果是？如果输入浮点数12345.6，结果会是怎样的？为什么？

#### 【微实例1.8】九九乘法表输出
九九乘法表输出。工整打印输出常用的九九乘法表，格式不限。

In [29]:
for i in range(1,10):
    for j in range(1,i+1):
        print("{}*{}={:2} ".format(j,i,i*j), end='')
    print('')

1*1= 1 
1*2= 2 2*2= 4 
1*3= 3 2*3= 6 3*3= 9 
1*4= 4 2*4= 8 3*4=12 4*4=16 
1*5= 5 2*5=10 3*5=15 4*5=20 5*5=25 
1*6= 6 2*6=12 3*6=18 4*6=24 5*6=30 6*6=36 
1*7= 7 2*7=14 3*7=21 4*7=28 5*7=35 6*7=42 7*7=49 
1*8= 8 2*8=16 3*8=24 4*8=32 5*8=40 6*8=48 7*8=56 8*8=64 
1*9= 9 2*9=18 3*9=27 4*9=36 5*9=45 6*9=54 7*9=63 8*9=72 9*9=81 


#### 扩展练习：你能够理解format括号里的{:2}吗？删掉最后一行代码会怎样？

#### 【微实例1.9】阶乘求和
计算 $1+2 !+3 !+\cdots+10 !$ 的结果

In [30]:
sum, tmp = 0, 1
for i in range(1,11):
    tmp*=i
    sum+=tmp
print("运算结果是: {}".format(sum))

运算结果是: 4037913


#### 扩展训练：你能够打印出来运算过程吗？
如：

1  =  1  +  0

3  =  2  +  1

9  =  6  +  3

33  =  24  +  9

153  =  120  +  33

873  =  720  +  153

5913  =  5040  +  873

46233  =  40320  +  5913

409113  =  362880  +  46233

4037913  =  3628800  +  409113

In [31]:
#请填充和优化注释行的代码
sum, tmp, sum_tmp = 0, 1, 0 #利用自定义的temp变量
for i in range(1,11):
    tmp*=i
    #sum_tmp = 
    sum+=tmp
    #print()
print("运算结果是: {}".format(sum))

运算结果是: 4037913


#### 【微实例1.10】猴子吃桃问题
猴子第一天摘下若干个桃子，当即吃了一半，还不过痕, 又多吃了一个: 第二天早上又将剩下的桃子吃掉一半，又多吃了一个。以后每天早 上都吃了前一天剩下的一半多一个。到第五天早上想再吃时，见只剩下一个桃子了。 请编写程序计算猴子第一天共摘了多少桃子。

In [32]:
n = 1
for i in range(5,0,-1):
    n = (n+1)<<1
print(n)

94


#### 【微实例1.11】健康食谱输出
健康食谱输出。列出 5 种不同食材，输出它们可能组成的所有菜式名称。

In [33]:
diet = ['西红柿', '花椰菜', '黄瓜', '牛排', '虾仁']
for x in range(0, 5):
    for y in range(0, 5):
        if not(x == y):
            print("{}{}".format(diet[x], diet[y]))

西红柿花椰菜
西红柿黄瓜
西红柿牛排
西红柿虾仁
花椰菜西红柿
花椰菜黄瓜
花椰菜牛排
花椰菜虾仁
黄瓜西红柿
黄瓜花椰菜
黄瓜牛排
黄瓜虾仁
牛排西红柿
牛排花椰菜
牛排黄瓜
牛排虾仁
虾仁西红柿
虾仁花椰菜
虾仁黄瓜
虾仁牛排


#### 扩展训练：

每天吃这些菜太单调了啊喂，可以来点新花样吗？你可以打印出你最喜欢的菜式组合吗？有些组合比如黄瓜虾仁，虾仁黄瓜，好像没什么区别欸！你知道怎么去掉吗？

#### 【微实例1.12】五角星绘制

In [34]:
from turtle import *
fillcolor("red")
begin_fill()
while True:
    forward(200)
    right(144)
    if abs(pos()) < 1:
        break
end_fill()

#### 【微实例1.13】太阳花绘制

In [35]:
from turtle import *
color('red', 'yellow')
begin_fill()
while True:
    forward(200)
    left(170)
    if abs(pos()) < 1:
       break
end_fill()
done()

#### 扩展训练：画个皮卡丘，观察箭头的方向
直接借鉴这位博主的代码，跑通就行哦！也可以用其他网上的代码，但记得一定要在注释上引用原文链接，这是基本准则！！

In [ ]:
#版权声明：本文为CSDN博主「冰柚子」的原创文章，遵循CC 4.0 BY-SA版权协议，转载请附上原文出处链接及本声明。
''' 原文链接：
    https://blog.csdn.net/qq_45110581/article/details/105025463
'''


## 练习题

### 1. 距离转化：
通常英里(miles)和公里(kilometers)的计算方式是约等于 

kilometers = (miles * 8.0) / 5.0

你能计算出输入英里，然后输出公里吗？反之，你能够输入公里，输出英里吗？
一般来说，数值是浮点数。

### 2.温度转换

温度的刻画有两个不同体系：摄氏度（Celsius）和华氏度（Fahrenheit）。‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬

请编写程序将用户输入华氏度转换为摄氏度，或将输入的摄氏度转换为华氏度。‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬

转换算法如下：（C表示摄氏度、F表示华氏度)

     C = ( F - 32 ) / 1.8‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬

     F = C * 1.8 + 32
     
#### 要求如下：‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬

(1) 输入输出的摄氏度可采用大小写字母C结尾，温度可以是整数或小数，如：12.34C指摄氏度12.34度；‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬

(2) 输入输出的华氏度可采用大小写字母F结尾，温度可以是整数或小数，如：87.65F指摄氏度87.65度；‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬

(3) 输出保留小数点后两位，输入格式错误时，输出提示：输入格式错误；‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬

(4) 使用input()获得测试用例输入时，不要增加提示字符串。

#### 这是本课程课上的实例，请注意两点：

(1) 将输入字符串转换为数字时使用eval()函数，不要用int()函数，因为输入的数字可能不是整数；

(2) 采用{:.2f}将输出数字变成两位小数点表示时，即使数学上该输出值是整数，也会按照小数方式输出，例如，转换后温度为10度，输出为10.00。


### 3. 温度转化2

温度的刻画有两个不同体系：摄氏度（Celsius）和华氏度（Fahrenheit）。‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬

请编写程序将用户输入华氏度转换为摄氏度，或将输入的摄氏度转换为华氏度。‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬

转换算法如下：（C表示摄氏度、F表示华氏度）‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬

         C = ( F - 32 ) / 1.8‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬

         F = C * 1.8 + 32‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬

#### 要求如下：‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬

(1) 输入输出的摄氏度采用大写字母C开头，温度可以是整数或小数，如：C12.34指摄氏度12.34度；‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬

(2) 输入输出的华氏度采用大写字母F开头，温度可以是整数或小数，如：F87.65指摄氏度87.65度；‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬

(3) 不考虑异常输入的问题，输出保留小数点后两位；‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬

(4) 使用input()获得测试用例输入时，不要增加提示字符串。


### 4. Hello World的条件输出
#### 描述：
获得用户输入的一个整数，参考该整数值，打印输出"Hello World"，要求：‪‬‪‬‪‬‪‬‪‬‮‬‫‬‭‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬

如果输入值是0，直接输出"Hello World"‪‬‪‬‪‬‪‬‪‬‮‬‫‬‭‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬

如果输入值大于0，以两个字符一行方式输出"Hello World"（空格也是字符）
‪‬‪‬‪‬‪‬‪‬‮‬‫‬‭‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬‬
如果输入值小于0，以垂直方式输出"Hello World"

#### 输入输出示例：

输入： 0

输出： Hello World

### 5. 数字形式转换 I

获得用户输入的一个正整数输入，输出该数字对应的中文字符表示。‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬

0到9对应的中文字符分别是：零一二三四五六七八九

### 6. 数值运算
#### 描述:
获得用户输入的一个字符串，格式如下：‪‬‪‬‪‬‪‬‪‬‮‬‫‬‭‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

M OP N‪‬‪‬‪‬‪‬‪‬‮‬‫‬‭‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

其中，M和N是任何数字，OP代表一种操作，表示为如下四种：+, -, *, /（加减乘除）‪‬‪‬‪‬‪‬‪‬‮‬‫‬‭‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

根据OP，输出M OP N的运算结果，统一保存小数点后2位。‪‬‪‬‪‬‪‬‪‬‮‬‫‬‭‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

注意：M和OP、OP和N之间可以存在多个空格，不考虑输入错误情况。‪‬‪‬‪‬‪‬‪‬‮‬‫‬‭‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

#### 输入输出示例：
示例 1	

输入：10 + 100 

输出：110.00 

示例 2

输入：1 / 20

输出： 0.05

### 7. 货币转换 I 
人民币和美元是世界上通用的两种货币之一，写一个程序进行货币间币值转换，其中：‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬

人民币和美元间汇率固定为：1美元 = 6.78人民币。‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬

程序可以接受人民币或美元输入，转换为美元或人民币输出。人民币采用RMB表示，美元USD表示，符号和数值之间没有空格。‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬

注意：‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬

(1) 这是一个OJ题目，获得输入请使用input() ；‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‫‬

(2) 不提示输出格式错误，结果小数点后保留两位。
